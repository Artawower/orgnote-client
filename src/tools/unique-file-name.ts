export const getUniqueFileName = (
  existingFileNames: string[],
  fileExt = '.org',
  filePrefix = 'untitled-note'
): string => {
  let initialName = `${filePrefix}${fileExt}`;

  let inc = 0;
  while (existingFileNames.includes(`${initialName}`)) {
    inc++;
    initialName = `${filePrefix}-${inc}${fileExt}`;
  }

  return initialName;
};
