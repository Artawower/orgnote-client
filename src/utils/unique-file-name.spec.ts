import { expect, test } from 'vitest';
import { getUniqueFileName } from './unique-file-name';

test('should return unique name', () => {
  const fileName = getUniqueFileName(['Untitled-note.org']);

  expect(fileName).toMatchSnapshot();
});

test('Should return unique name with multiple default names', () => {
  const fileNames = ['Untitled-note.org', 'Untitled-note-1.org', 'Untitled-note-2.org'];
  const fileName = getUniqueFileName(fileNames);
  expect(fileName).toMatchSnapshot();
});
