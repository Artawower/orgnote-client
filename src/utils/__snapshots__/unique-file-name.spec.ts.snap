// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should return unique name with multiple default names 1`] = `"untitled-note.org"`;

exports[`should return unique name 1`] = `"untitled-note.org"`;
